src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIAssistantClient.cs

```diff
@@ -0,0 +1,350 @@
+                    break;
+
+                case RequiredChatToolMode required:
+                    runOptions.ToolConstraint = required.RequiredFunctionName is null ?
+                        new ToolConstraint(ToolDefinition.CreateFunction(required.RequiredFunctionName)) :
+                        ToolConstraint.Required;
+                    break;
+            }
+
+            // Store the response format.
+            if (options.ResponseFormat is ChatResponseFormatText)
+            {
+                runOptions.ResponseFormat = AssistantResponseFormat.Text;
+            }
+            else if (options.ResponseFormat is ChatResponseFormatJson jsonFormat)
+            {
+                runOptions.ResponseFormat = jsonFormat.Schema is { } jsonSchema ?
+                    AssistantResponseFormat.CreateJsonSchemaFormat(
+                        jsonFormat.SchemaName ?? "json_schema",
+                        BinaryData.FromBytes(JsonSerializer.SerializeToUtf8Bytes(jsonSchema, OpenAIJsonContext.Default.JsonElement)),
+                        jsonFormat.SchemaDescription) :
+                    AssistantResponseFormat.JsonObject;
+            }
+        }
+
+        // Handle ChatMessages. System messages are turned into additional instructions.
+        StringBuilder? instructions = null;
+        List<FunctionResultContent>? functionResults = null;
+        foreach (var chatMessage in chatMessages)
+        {
+            List<MessageContent> messageContents = [];
+
+            if (chatMessage.Role == ChatRole.System)
+            {
+                instructions ??= new();
+                foreach (var textContent in chatMessage.Contents.OfType<TextContent>())
+                {
+                    _ = instructions.Append(textContent);
+                }
+
+                continue;
+            }
+
+            foreach (AIContent content in chatMessage.Contents)
+            {
+                switch (content)
+                {
+                    case TextContent tc:
+                        messageContents.Add(MessageContent.FromText(tc.Text));
+                        break;
+
+                    case DataContent dc when dc.MediaTypeStartsWith("image/"):
+                        messageContents.Add(MessageContent.FromImageUri(new(dc.Uri)));
+                        break;
+
+                    case FunctionResultContent frc:
+                        (functionResults ??= []).Add(frc);
+                        break;
+                }
+            }
+
+            if (messageContents.Count > 0)
+            {
+                runOptions.AdditionalMessages.Add(new(
+                    chatMessage.Role == ChatRole.Assistant ? MessageRole.Assistant : MessageRole.User,
+                    messageContents));
+            }
+        }
+
+        if (instructions is not null)
+        {
+            runOptions.AdditionalInstructions = instructions.ToString();
+        }
+
```

Comment:
I no longer clear the user's history; the need for that is gone with the new semantics around {Streaming}ChatCompletion{Update}.ChatThreadId.